std/math.d

```diff
@@ -4808,6 +4808,7 @@ unittest
     }
     ensureDefaults();
 
+    version(D_HardFloat)
```

Comment:
@smolt and @redstar, this assert throws on Android/ARM, which uses soft-float by default.  I don't think this test should be run for soft-float generally, any other architecture on which it should?  If so, I can disable it only for ARM soft-float.